Source/ParameterEncoder.swift

```diff
@@ -221,6 +224,45 @@ public final class URLEncodedFormEncoder {
         }
     }
 
+    /// Configures how `Date` parameters are encoded.
+    public enum DateEncoding {
+        /// Encodes the date from `Date` structure.
+        case deferredToDate
+        /// Encodes the date as a UNIX timestamp.
+        case secondsSince1970
+        /// Encodes the date as UNIX millisecond timestamp.
+        case millisecondsSince1970
+        /// Encodes the date as an ISO-8601-formatted string (in RFC 3339 format).
+        case iso8601
+        /// Encodes the date as a string formatted by the given formatter.
+        case formatted(DateFormatter)
+        /// Encodes the date as a custom string formatted by the given closure.
+        case custom((Date) -> String)
+
+        /// Encodes the date according to the strategy.
+        ///
+        /// - Parameter string: The `Date` to encode.
+        /// - Returns:          The encoded `String` or `nil` if the date should be encoded as `Encodable` structure.
+        func encode(_ value: Date) -> String? {
+            switch self {
+            case .deferredToDate:
+                return nil
+            case .secondsSince1970:
+                return String(value.timeIntervalSince1970)
+            case .millisecondsSince1970:
+                return String(value.timeIntervalSince1970 * 1000.0)
+            case .iso8601:
+                return ISO8601DateFormatter.string(from: value,
```

Comment:
Done.

Reply:
There is a comment in the [JSONEncoder source code](https://github.com/apple/swift/blob/b0f5815d2b003df628b1bcfe94681fec489c9492/stdlib/public/Darwin/Foundation/JSONEncoder.swift#L2553) explaining the reason for using the `var` property. It seems that JSONEncoder uses `var` in order for a property to be lazy. This is necessary so that this code does not run on an older OS that does not support `ISO8601DateFormatter` (macOS < 10.12, iOS < 10.0, watchOS < 3.0, tvOS < 10.0). There is no such problem in Alamofire, so I used `let` property. Please let me know if I need to change it to `var`.